### PR TITLE
User endpoints, test fixes

### PR DESCRIPTION
**Pull request by Celeo**
Changes:

- Fixed (hopefully) GitHub Actions not running
- Renamed JWT file for naming conventions
- Updated some variable names to better state their purpose
- Fixed DB connection while running tests so that we can actually assert on DB contents when running tests that call into the app via HTTP instead of just method calls
- Fixed endpoint routing prefixes
- Added docstrings to various methods
- Added endpoint for a logged-in user to update their own profile information, including password
- Added tests for user endpoints
- Fixed failing tests

### CODE REVIEW

File: backend/app/crud.py [hunk 1]
```diff
@@ -31,26 +31,26 @@ def get_users(db: Session, skip: int = 0, limit: int = 100) -> List[models.User]
...
+def create_user(db: Session, data: schemas.UserCreate):
     """Create a new user."""
-    password = user.user_password
+    password = data.user_password
     hashedpass = hash_password(password)
     db_user = models.User(
-        user_email=user.user_email,
-        user_first=user.user_first,
-        user_last=user.user_last,
-        user_hashed_password=str(hashedpass),
```
Comment (Celeo) [inline backend/app/crud.py]: If someone can explain why the cast to string was added, I'd be open to it - having this here, however, was failing tests.

Comment (petersteele111) [reply backend/app/crud.py]: So Pycharm was yelling at me that it was expecting a string for that variable but it was byte type instead. So I cast it to string and recast to bytes further down in another function.

File: backend/app/routers/user.py [hunk 2]
```diff
@@ -1,32 +1,72 @@
...
 
-@router.get("/users/", response_model=List[int])
+@router.get("/", response_model=List[int])
 def get_users(skip: int = 0, limit: int = 100, db: Session = Depends(get_db)):
+    """Endpoint to get a list of user ids."""
     return [user.user_id for user in crud.get_users(db, skip=skip, limit=limit)]
 
 
-@router.get("/users/{user_id}", response_model=schemas.User)
+@router.get("/{user_id}", response_model=schemas.User)
```
Comment (jmamcallister) [inline backend/app/routers/user.py]: Should this path start with "/users"?

Comment (jmamcallister) [reply backend/app/routers/user.py]: Ignore, didn't see the prefixes part before commenting :)

Comment (JLudwig2552) [reply backend/app/routers/user.py]: Everything inside routers/user.py is prefixed with "/users" see 
`app.include_router(user.router, prefix="/users")` in main.py

**Pull request merged**